kernel/kernel.ld

```diff
@@ -5,12 +5,15 @@ PAGE_SIZE = 4096;
 STACK_SIZE = PAGE_SIZE * 1;
 SECTIONS
 {
-    .text KERNEL_SEG : {
+    .text KERNEL_SEG : ALIGN(16) {
```

Comment:
check the alignment among debug / release from where the `main` is starting ... (it might need an alignment at section level maybe?)